add field values as enum

leojustino opened the pull request:
add field values in all clases to make reference to its values like an enum

leojustino committed:
add field values in all clases to make reference to its values like an enum

pd4d10 commented:
Hi @leojustino , thanks for your contribution!

Could you please explain the use cases?

BTW, the code is generated by `tool/generate.dart`. It would be appreciated if you could update it too.

leojustino committed:
new generated files

leojustino committed:
generalyze dartFormatter;
refactor to standard dart style;
use string buffer instead string concatenation;
change var to final where it wont change;
change bash command when it is windows;
generate field values as array with all other values of a class, like an enum;

leojustino commented:
lol sorry about the generation tool, i did not saw that;
what i did was generate an attribute in each class named values that contains a map referencing an string with each font icon, with this i accomplish to reference each field of this class with its string name that allow us to use it for search, also the field values make reference to all its icons (fields), just like works an enum, this makes easy access all vallues of the class programmatically without use of reflection.
